src/applications/apps.js

```diff
@@ -31,24 +30,20 @@ export function registerApplication(appName, arg1, arg2, customProps = {}) {
     throw new Error(`The first argument must be a non-empty string 'appName'`);
   if (getAppNames().indexOf(appName) !== -1)
     throw new Error(`There is already an app declared with name ${appName}`);
-  if (typeof customProps !== 'object')
+  if (typeof customProps !== 'object' || Array.isArray(customProps))
     throw new Error('customProps must be an object');
 
+  if (!arg1)
```

Comment:
Now that we aren't supporting two different ways of registering the application, we can rename `arg2` to be `activityFn` in the [function defn](https://github.com/CanopyTax/single-spa/pull/175/files#diff-09ec3011a4b92de4d61e29380e950bc3R28) (and in [this one](https://github.com/CanopyTax/single-spa/pull/175/files#diff-09ec3011a4b92de4d61e29380e950bc3R23), too).

The `arg1` also could be renamed if we wanted, to be something like ` appOrLoadingFn` if we wanted.

Reply:
I _almost_ did that... I'll fix it because I think it adds clarity.